OpenRA.Mods.Common/Traits/Immobile.cs

```diff
@@ -53,6 +53,9 @@ public Immobile(ActorInitializer init, ImmobileInfo info)
 
 		public void AddedToWorld(Actor self)
 		{
+			if (occupied == IOccupySpaceExts.NoOccupiedCells)
+				return;
+
 			self.World.ActorMap.AddInfluence(self, this);
 			self.World.ActorMap.AddPosition(self, this);
 			self.World.ScreenMap.Add(self);
```

Comment:
`ScreenMap.Add` uses the actor bounds rather than the footprint.  The size of the bounds is completely disconnected from the footprint defined in this trait.